athena-udfs/src/test/java/com/amazonaws/athena/connectors/udfs/AthenaUDFHandlerTest.java

```diff
@@ -141,45 +130,41 @@ public void testKmsEncryption() throws Exception
     @Test
     public void testKmsDecryptionEndToEnd() throws Exception
     {
-        String secretName = "<fill-in-your-secret-name>";
+        String secretName = "<fill-in-secret-name>";
         String secretValue = "<fill-in-secret-value>";
 
         this.athenaUDFHandler = new AthenaUDFHandler();
 
-        SecretKeySpec skeySpec = new SecretKeySpec(decoder.decode(secretValue), "AES");
-        Cipher cipher = Cipher.getInstance("AES");
-        cipher.init(Cipher.ENCRYPT_MODE, skeySpec);
+        Cipher cipher = AthenaUDFHandler.getCipher(Cipher.ENCRYPT_MODE, decoder.decode(secretValue.getBytes()), AthenaUDFHandler.getGCMSpecEncryption());
 
         String expected = "abcdef";
-        String encryptedString = new String(encoder.encode(cipher.doFinal(expected.getBytes())));
+        byte[] encryptedString = cipher.doFinal(expected.getBytes(StandardCharsets.UTF_8));
+        ByteBuffer byteBuffer = ByteBuffer.allocate(AthenaUDFHandler.GCM_IV_LENGTH + encryptedString.length);
+        byteBuffer.put(cipher.getIV());
+        byteBuffer.put(encryptedString);
+        String encodedString = new String(encoder.encode(byteBuffer.array()));
 
-        String result = athenaUDFHandler.decrypt(encryptedString, secretName);
+        String result = athenaUDFHandler.decrypt(encodedString, secretName);
 
         assertEquals(expected, result);
     }
 
     /**
-     * This UT is used to test {@link AthenaUDFHandler#encrypt(String, String)} method end-to-end.
+     * This UT is used to test encryption and decryption end-to-end.
      * It requires AWS Secret Manager setup and AWS credential setup.
      * @throws Exception
      */
     @Ignore("Enabled as needed to do end-to-end test")
     @Test
-    public void testKmsEncryptionEndToEnd() throws Exception
```

Comment:
This test was identical to `testKmsDecryptionEndToEnd` so I repurposed it to test both methods end-to-end.